### DISCUSSION TITLE
Pairing order enddevice vs router?

### DISCUSSION THREAD

**Discussion by JacekKac** [Koenkk/zigbee2mqtt#14112] [General]
Hi there, 
what is the best way to create a network with enddevices and repeaters in case of pairing order? Is the end device route choosen automaticly or end device has to be paired with router which is paired with controler? Not sure if i can just pair all devices and zigbee2mqtt firmware will do the rest or should i be more specific with the order?

**Comment by DubhAd** [comment]
Start with the routers, then once you've done the routers do the end-devices.

Start by the coordinator and spiral out each time.

Z2M has no say in how devices route through the mesh, though at pairing time you can force them to join via a particular router (or the coordinator). This is particularly useful for Xiaomi devices (at least the Zigbee 1.2 ones) since they'll pair via the first device they hear from, and will tend to stay paired that way.

**Reply by JacekKac** [reply]
1/ tried in a complete new environment, it seems that TUYA termostat [TS0201](https://www.zigbee2mqtt.io/devices/TS0201.html#tuya-ts0201) **never uses a router** to join (sonoff wall plug R2ZB) even if the coordinator is not in range. Router was the only deviced with alow join.

i pair them close to the coordinator and then moved them to other floors (where wall plugs are installed). Base on MAP it seems that they are using wallplags to get to the controller. Strange but working

![image](https://user-images.githubusercontent.com/72306272/191609598-cb1887ad-f0fc-48f0-bd3c-985def9de078.png)

**Comment by JacekKac** [comment]
My current network MAP. Temp_10 and Temp_11 are the only one not working rest is sending reports, but some of them are not linked on a map? Is it correct... ? dont'know what to do at this point. Problematic connection is between WallPlug1,WallPLug2 and zgibemini 

![image](https://user-images.githubusercontent.com/72306272/192093214-b40c051a-542f-4077-bb95-9e446b84d5eb.png)
